src/agor/tools/git_operations.py

```
@@ -0,0 +1,220 @@
+    try:
+        # Detect git binary
+        git_binary = "/bin/git"  # Default for most systems
+        if not Path(git_binary).exists():
+            # Fallback detection
+            result = subprocess.run(
+                ["which", "git"], capture_output=True, text=True, timeout=10
+            )
+            if result.returncode == 0:
+                git_binary = result.stdout.strip()
+            else:
+                git_binary = "git"  # Hope it's in PATH
```

Ruff (0.11.9)

Use ternary operator `git_binary = result.stdout.strip() if result.returncode == 0 else "git"` instead of `if`-`else`-block

(SIM108)